### PR TITLE
Bump globals.yml .NET SDK version from 3.1.405 to 6.x + assorted changes

### PR DESCRIPTION
**Pull request by konrad-jamrozik**
## Context

Updating .NET SDK version from `3.1.405` to `6.x` for our pipelines because:

- `.NET Core 3.1` support [is ending on December 13, 2022](https://dotnet.microsoft.com/en-us/download/dotnet), hence we want to move to a supported SDK. See also #4934.
- As `.NET Core 3.1` support is ending, [it got removed from latest Ubuntu images](https://github.com/Azure/azure-sdk-tools/issues/4888#issuecomment-1342033218), and in some places we we didn't pin the Ubuntu image, thus resulting in us using the latest image, thus leading to issues, like #4888.
- To address #4888 we need to move `PipelineGenerator` to .NET 6, which we do in PR #4915. But for the PR #4915 to successfully build, this PR needs to be merged first, as explained in https://github.com/Azure/azure-sdk-tools/pull/4915#issuecomment-1342167140.
- Updated `GitHubIssues` `TargetFramework` as it wasn't included in update made in PR #4937.

I chose `6.x` because this is the version installed on the image we use, as explained in https://github.com/Azure/azure-sdk-tools/pull/4915#issue-1483843025. 

Note: Alternatively, I could have used `6.0.403`, because it is a string that is present in `releases / sdk / version` in the [releases.json](https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/6.0/releases.json), as explained in [UseDotNet@2 task doc](https://learn.microsoft.com/en-gb/azure/devops/pipelines/tasks/reference/use-dotnet-v2?view=azure-pipelines&viewFallbackFrom=azure-devops). For more, see https://github.com/Azure/azure-sdk-tools/pull/4915#issuecomment-1342167140.

## Changes made

- Primarily, the .NET SDK version in `globals.yml`, which is picked by our `archetype` files;
- Also changed `archetype-sdk-tool-azure-function.yml` to use `net6.x` and does few other assorted changes. 
    - Updated used image from Ubuntu 18.04, 1ES-hosted Ubuntu 20.04.
- Got rid of temporary project-specific .NET overrides.
- In `GitHubIssues`, updated `Microsoft.NET.Sdk.Functions` from `3.0.9` to `4.1.3`, as the old version depended on obsolete .NET SDK version.
- In `archetype-sdk-tool-dotnet.yml`, ensured the directory from which packages are to be published always exists, even if it wasn't created by previous steps because they didn't create any packages. This is the case e.g. for `identity-resolution - ci` which caused it to fail after the update of .NET version. This appears to be a behavior change introduced in .NET 6 or earlier. Unfortunately, there is no "directory exists" expression: the doc doesn't mention it, and relevant issues or feature requests only propose workarounds: [azure-devops-docs #1877](https://github.com/MicrosoftDocs/azure-devops-docs/issues/1877?) and [developer community 366095](https://developercommunity.visualstudio.com/t/add-an-exists-on-task-custom-condition/366095).

Note I **did not** update `CheckEnforcer` and `WebhookRouter` as they are scheduled for deletion per https://github.com/Azure/azure-sdk-tools/pull/4937#pullrequestreview-1214475397. PRs doing that: 
- #4963
- #4966

##

### CODE REVIEW

File: eng/pipelines/templates/variables/globals.yml
Comment (weshaggard) [inline eng/pipelines/templates/variables/globals.yml:4]: We are also going to need to update our target frameworks in the various projects.

Comment (weshaggard) [reply eng/pipelines/templates/variables/globals.yml:4]: We have also talked about moving this version into https://github.com/Azure/azure-sdk-tools/blob/main/global.json similar to what azure-sdk-for-net repo does.

Comment (konrad-jamrozik) [reply eng/pipelines/templates/variables/globals.yml:4]: @weshaggard 

> We are also going to need to update our target frameworks in the various projects.

I found more issues related to this caused by SDKs going out of support. Please see #4934 and #4935.

However, once these problems are solved, I don't think that `TargetFramework` has to exactly match the SDK used, as long as the targeted framework is not newer than the SDK and is not obsolete. Which admittedly is a bit moot point now, because everything before `net6.0` is obsolete (-:

> We have also talked about moving this version into https://github.com/Azure/azure-sdk-tools/blob/main/global.json similar to what azure-sdk-for-net repo does.

How about I do the migration to `global.json` in a separate, follow-up PR?

Also, I think this PR is no longer required to unblock PR [#4915 per my comment there](https://github.com/Azure/azure-sdk-tools/pull/4915#issuecomment-1345059377), so I suggest we go ahead with merging #4915 to mitigate #4888, and then work on more broad fix proposed by this PR.

**Comment by konrad-jamrozik** [comment]
@weshaggard @benbp this PR is ready for review for approval. Thx! :)

**Review by benbp — APPROVED** [review]
Thanks @konrad-jamrozik !

File: tools/code-owners-parser/ci.yml
Comment (weshaggard) [inline tools/code-owners-parser/ci.yml:29]: Can these test steps be re-enabled?

**Review by weshaggard — APPROVED** [review]
The changes look reasonable to be once CI is passing.

File: tools/code-owners-parser/ci.yml
Comment (konrad-jamrozik) [reply tools/code-owners-parser/ci.yml:29]: I will enable them in a separate PR. I have a work item for it: https://github.com/Azure/azure-sdk-tools/issues/4938. I believe I need to make the script [refer to a new package](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/get-codeowners.ps1#L4), which I will will build and refer to once this PR is merged (even though I should be able to update it even before merging the PR; just being extra-safe here).

File: eng/pipelines/templates/stages/archetype-sdk-tool-dotnet.yml
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-sdk-tool-dotnet.yml:95]: Might actually be worth doing a directory listing so we know what is in there to help debugging potential future issues. 

Also probably worth capturing the `$(Build.ArtifactStagingDirectory)/packages` in a variable and using that variable in all the places we access that in this file.

Comment (konrad-jamrozik) [reply eng/pipelines/templates/stages/archetype-sdk-tool-dotnet.yml:95]: @weshaggard addressed by:
- #5007

**Pull request merged**
